mmcv/runner/base_runner.py

```diff
@@ -306,6 +306,28 @@ def call_hook(self, fn_name):
         for hook in self._hooks:
             getattr(hook, fn_name)(self)
 
+    def get_hook_info(self):
+        # Get hooks info in each stage
+        stage_hook_map = {stage: [] for stage in Hook.stages}
+        for hook in self.hooks:
+            try:
+                priority = Priority(hook.priority).name
+            except ValueError:
+                priority = hook.priority
+            classname = hook.__class__.__name__
+            hook_info = f'({priority:<12}) {classname:<35}'
+            for trigger_stage in hook.get_triggered_stages():
+                stage_hook_map[trigger_stage].append(hook_info)
+
+        stage_hook_infos = []
+        for stage in Hook.stages:
+            hook_infos = stage_hook_map[stage]
+            if len(hook_infos) > 0:
+                info = '{}:\n{}\n'.format(stage, '\n'.join(hook_infos))
+                info += ' -------------------- '
+                stage_hook_infos.append(info)
```

Comment:
Are they sorted?

Reply:
Both stages and hooks are sorted.

Reply:
Where are hooks in stages sorted?

Reply:
Here:
https://github.com/open-mmlab/mmcv/blob/94818ad1368998a19781aca8460a2d3a2cab322d/mmcv/runner/base_runner.py#L255-L281